### PR TITLE
Add RIRO and Static masks to realtime zshimming

### LINKED ISSUE #220: Add masks for RIRO and Static in realtime_zshimming
### Description of Problem / Inefficiency:
With the goal of improving our realtime zshimming workflow, we can enhance our masking capability. We currently mask the shimming region we want to improve. Adding the ability to mask the riro component and the static component independently would be a nice feature and a lead to further investigations.

### Implementation Suggestion:
Add an option that inputs a mask for RIRO and use the current one for static (or vice versa)

### PR DESCRIPTION
**Pull request by po09i**
## Description
With the goal of improving our realtime zshimming workflow, we can enhance our masking capability. We currently mask the shimming region we want to improve. Adding the ability to mask the riro component and the static component independently would be a nice feature and a lead to further investigations.

This PR adds `nii_mask_anat_static` and `nii_mask_anat_riro` to the `realtime_zshim()` API and appropriately masks the relevant components. The PR also adds `mask_riro` and `mask_static` options to the `st_realtime_zshim` CLI. The relevant tests were modified and the demo shell script was updated.

## Linked issues
Fixes #220

### CODE REVIEW

**Comment by evaalonsoortiz** [comment]
I am trying to test this out an getting the following error:

```
The filename for the output mask is: /Users/evaalonsoortiz/Documents/Academic/Postdoc_Julien/Projects/Realtime_zShimming/Data/acdc_123p/rt_zshim_nifti/sub-example/fmap/sub-example_anat_mask.nii.gz
Usage: st_realtime_zshim [OPTIONS]
Try 'st_realtime_zshim -h' for help.

Error: no such option: -m
```

Note: I'm not using -m as an option in the call to `st_realtime_zshim`

**Comment by po09i** [comment]
Can you post the shell call you are using so I can try to reproduce the error

**Comment by evaalonsoortiz** [comment]
> Can you post the shell call you are using so I can try to reproduce the error

`st_prepare_fieldmap "sub-example_phasediff.nii.gz" -mag "sub-example_magnitude1.nii.gz" -unwrapper "prelude" -output "sub-example_fieldmap.nii.gz" -gaussian_filter=True || exit`

`st_mask box -input "../anat/sub-example_unshimmed_e1.nii.gz" -size 20 20 30 -center 60 32 7 -output "sub-example_anat_mask.nii.gz" || exit
`

`st_realtime_zshim -fmap "sub-example_fieldmap.nii.gz" -anat "../anat/sub-example_unshimmed_e1.nii.gz" -resp "../../../realtime_zshimming_data/PMUresp_signal.resp" -mask_anat "sub-example_anat_mask.nii.gz" -mask_riro "sub-example_anat_mask.nii.gz" -output "." || exit
`

**Comment by jcohenadad** [comment]
@evaalonsoortiz no need for the "" in the syntax (quicker to type without them 😊 )

**Comment by po09i** [comment]
@evaalonsoortiz There was a typo in the shell script, the option was -mask_anat but it should have been -mask_static. I corrected it and it should work now.

**Comment by jcohenadad** [comment]
i didn't realize there was a mix of "-" and "_" in CLI flags. We should stick to "-" only-- can someone pls open an issue @charjl @po09i

File: shimmingtoolbox/cli/realtime_zshim.py [hunk 1]
```diff
@@ -22,13 +22,16 @@
               help="Filename of the anatomical image to apply the correction.")
 @click.option('-resp', 'fname_resp', type=click.Path(), required=True,
               help="Siemens respiratory file containing pressure data.")
-@click.option('-mask', 'fname_mask_anat', type=click.Path(), required=False,
-              help="3D nifti file with used to define the spatial region to shim. "
+@click.option('-mask_static', 'fname_mask_anat_static', type=click.Path(), required=False,
+              help="3D nifti file used to define the static spatial region to shim. "
+                   "The coordinate system should be the same as ``anat``'s coordinate system.")
+@click.option('-mask_riro', 'fname_mask_anat_riro', type=click.Path(), required=False,
+              help="3D nifti file used to define the time varying spatial region to shim. "
```
Comment (jcohenadad) [inline shimmingtoolbox/cli/realtime_zshim.py]: spell out RIRO in the help

File: shimmingtoolbox/cli/realtime_zshim.py [hunk 2]
```diff
@@ -22,13 +22,16 @@
               help="Filename of the anatomical image to apply the correction.")
 @click.option('-resp', 'fname_resp', type=click.Path(), required=True,
               help="Siemens respiratory file containing pressure data.")
-@click.option('-mask', 'fname_mask_anat', type=click.Path(), required=False,
-              help="3D nifti file with used to define the spatial region to shim. "
+@click.option('-mask_static', 'fname_mask_anat_static', type=click.Path(), required=False,
```
Comment (jcohenadad) [inline shimmingtoolbox/cli/realtime_zshim.py]:
```suggestion
@click.option('-mask-static', 'fname_mask_anat_static', type=click.Path(), required=False,
```

File: shimmingtoolbox/cli/realtime_zshim.py [hunk 3]
```diff
@@ -22,13 +22,16 @@
...
+              help="3D nifti file used to define the static spatial region to shim. "
+                   "The coordinate system should be the same as ``anat``'s coordinate system.")
+@click.option('-mask_riro', 'fname_mask_anat_riro', type=click.Path(), required=False,
```
Comment (jcohenadad) [inline shimmingtoolbox/cli/realtime_zshim.py]:
```suggestion
@click.option('-mask-riro', 'fname_mask_anat_riro', type=click.Path(), required=False,
```

**Comment by evaalonsoortiz** [comment]
> @evaalonsoortiz There was a typo in the shell script, the option was -mask_anat but it should have been -mask_static. I corrected it and it should work now.

Just gave it another go, but still getting the same error ...

**Comment by po09i** [comment]
> > @evaalonsoortiz There was a typo in the shell script, the option was -mask_anat but it should have been -mask_static. I corrected it and it should work now.
> 
> Just gave it another go, but still getting the same error ...

Problem is now fixed. The error was the one described above [here](https://github.com/shimming-toolbox/shimming-toolbox/pull/221#issuecomment-774254852)

**Review by evaalonsoortiz — APPROVED** [review]
Tested it out, looks good.

**Pull request merged**